core/src/test/java/kafka/test/MockController.java

```diff
@@ -181,10 +194,16 @@ public CompletableFuture<Void> unregisterBroker(
     static class MockTopic {
         private final String name;
         private final Uuid id;
+        private final int numPartitions;
 
         MockTopic(String name, Uuid id) {
+            this(name, id, 10); // hard-code the number of partitions if left unspecified
```

Comment:
I guess this is a nitpick but `num.partitions` defaults to 1, so having a default of 10 partitions is a bit surprising here. Is there a reason to use 10 instead of 1?